BootloaderCommonPkg/Library/IppCryptoLib/auth/pcpsha512ca.c

```diff
@@ -560,6 +548,39 @@ void UpdateSHA512(void* uniHash, const Ipp8u* mblk, int mlen, const void* uniPra
       digest[7] += v[7];
    }
 }
+
+/*F*
+//    Name: UpdateSHA512
+//
+// Purpose: Update internal hash according to input message stream.
+//
+// Parameters:
+//    uniHash  pointer to in/out hash
+//    mblk     pointer to message stream
+//    mlen     message stream length (multiple by message block size)
+//    uniParam pointer to the optional parameter
+//
+*F*/
+
+void UpdateSHA512(void* uniHash, const Ipp8u* mblk, int mlen, const void* uniPraram)
+{
+#if defined(_SLIMBOOT_OPT)
+
+   if (FixedPcdGet32 (PcdCryptoShaOptEnabled) & IPP_CRYPTO_SHA384_W7) {

```

Comment:
Same here, maybe keep W7 as default.   If no bit is set in mask,  you will either need to have a default implementation or provide error for compiling.